primefaces/src/main/java/org/primefaces/component/datagrid/DataGrid.java

```diff
@@ -86,10 +86,17 @@ public void loadLazyData() {
         if (model instanceof LazyDataModel) {
             LazyDataModel lazyModel = (LazyDataModel) model;
 
-            lazyModel.setRowCount(lazyModel.count(Collections.emptyMap()));
-            calculateFirst();
-
+            if (getFirst() > 0) {
```

Comment:
We should put `getFirst()` into a local variable so the getter is not called multiple times.

Reply:
It's a bit tricky because first will change in calculateFirst. Maybe calculateFirst should return first and in the only case that the boolean value is checked (the DataTable one), store the old first and compare it. 

WDYT?

Reply:
No you can leave as is